### PR TITLE
Little code cleanup

### PR DESCRIPTION
**Pull request by arthurzam**
Some little code optimizations, readability improvement and shortening.

### CODE REVIEW

File: src/modules/FFmpeg/FFDecSW.cpp [hunk 1]
```diff
@@ -339,8 +339,6 @@ bool FFDecSW::open(StreamInfo &streamInfo, VideoWriter *)
 		return false;
 	if (codec_ctx->codec_type == AVMEDIA_TYPE_VIDEO)
 	{
-		if (codec->capabilities & CODEC_CAP_DR1)
```
Comment (zaps166) [inline src/modules/FFmpeg/FFDecSW.cpp]: IIRC this change will produce incorrect frames (with borders?) for FFmpeg 2.2 which (I hope) still works.

Comment (arthurzam) [inline src/modules/FFmpeg/FFDecSW.cpp]: As I checked with FFmpeg 2.2 the CODEC_FLAG_EMU_EDGE flag was always set. Maybe I'm missing something?

Comment (arthurzam) [inline src/modules/FFmpeg/FFDecSW.cpp]: look at this [link](https://github.com/FFmpeg/FFmpeg/blob/n2.2/libavcodec/avcodec.h#L738)

Comment (zaps166) [inline src/modules/FFmpeg/FFDecSW.cpp]: > Maybe I'm missing something?

Yes, there is at lease one difference between FFmpeg 2.2 and 2.3 in usage of this flag, see: https://github.com/zaps166/QMPlay2/commit/d657f5ca409a621871e762c9f10e53f0de3659fa

Comment (arthurzam) [inline src/modules/FFmpeg/FFDecSW.cpp]: I understand - now will revert this patch.

Comment (zaps166) [inline src/modules/FFmpeg/FFDecSW.cpp]: > ... I.e. this flag is now always set.

So this looks like FFmpeg 2.2.x bug :) The best way is to compile & test :smiley:

Comment (arthurzam) [inline src/modules/FFmpeg/FFDecSW.cpp]: I will leave this one to future cleanup pull request, after many more checks...

File: src/modules/FFmpeg/VAAPIWriter.hpp [hunk 2]
```diff
@@ -27,7 +27,6 @@
 
 class VAAPIWriter : public QWidget, public VideoWriter
 {
-	Q_OBJECT
```
Comment (zaps166) [inline src/modules/FFmpeg/VAAPIWriter.hpp]: This class has slots.

File: src/modules/FFmpeg/VDPAUWriter.hpp [hunk 3]
```diff
@@ -31,7 +31,6 @@ struct _XDisplay;
 
 class VDPAUWriter : public QWidget, public VideoWriter
 {
-	Q_OBJECT
```
Comment (zaps166) [inline src/modules/FFmpeg/VDPAUWriter.hpp]: This class has slots.

File: src/modules/FFmpeg/VAAPIWriter.hpp (hunk 2 above)
Comment (arthurzam) [inline src/modules/FFmpeg/VAAPIWriter.hpp]: Fixed

File: src/modules/FFmpeg/VDPAUWriter.hpp (hunk 3 above)
Comment (arthurzam) [inline src/modules/FFmpeg/VDPAUWriter.hpp]: Fixed

File: src/gui/AddressDialog.hpp [hunk 4]
```diff
@@ -22,11 +22,12 @@
 #include <AddressBox.hpp>
 
 #include <QCheckBox>
+#include <QCoreApplication>
```
Comment (zaps166) [inline src/gui/AddressDialog.hpp]: If include order doesn't matter, please group them (if necessary, eg. QMPlay2 includes, Qt includes, C includes) and sort from the longest to the shortest (in new code) :D

In this case:
```
#include <QCoreApplication>
#include <QCheckBox>
#include <QDialog>
```

**Comment by arthurzam** [comment]
> I should review it carefully to not get regressions - code cleanup can do this :)

No problem, as I agree with it.

**Comment by zaps166** [comment]
Hmm, it works. Let's hope that it doesn't provide difficult-to-debug regressions :)

**Pull request merged**